Core/Src/Scheduler/Scheduler.c

```diff
@@ -11,23 +11,21 @@
  * Initializes the priority queue and schedules tasks based on the given sensors
  * and their update frequencies. Limits the number of sensors to MAX_SENSORS.
  */
-void SchedulerInit(Scheduler* scheduler, Sensor* sensorArray[], int numSensors) {
-    if (numSensors > MAX_SENSORS) {
-        printf("Warning: Number of sensors exceeds MAX_SENSORS. Some sensors will not be scheduled.\n");
-        numSensors = MAX_SENSORS;  // Limit the number of sensors to MAX_SENSORS
-    }
-
+void SchedulerInit(Scheduler* scheduler, Updateable* updatableArray[]) {
     PQInit(&scheduler->tasks);
     scheduler->running = false;
 
-    for (int i = 0; i < numSensors; i++) {
-        Task task;
-        Sensor* sensor = sensorArray[i];
-        if (sensor == NULL) {
-            continue;
+    int count = 0;
+    for (Updateable* updateable; updateable != NULL; updateable++) {
```

Comment:
1. The updateable pointer is declared as a pointer but not initialized with an intial value.
It is just a random pointer that may just contain whatever happened to be in that memory location.

2. The updatableArray was passed to the function, but this loop never reads from it. 
So the loop may just incrementing a random pointer without ever looking at the array elements.

Suggested code:

```
for (int i = 0; updatableArray[i] != NULL; i++) {
        if (count >= MAX_SENSORS) {
            printf("Warning: Number of sensors exceeds MAX_SENSORS. Some sensors will not be scheduled.\n");
            break;
        }

        Updateable* updateable = updatableArray[i];
        if (!updateable->hz) continue;  // Skip invalid frequencies (if it exceeds MAX_HZ)

        Task task;
        ...
```